### DISCUSSION TITLE
Unhandled exception.

### DISCUSSION THREAD

**Discussion by david-cueto** [ORelio/Minecraft-Console-Client#2613] [Q&A] [answered]
I cannot log in to a server: (running it in a VPS running ubuntu 22.04)

You chose to run in offline mode.
Server IP :
Resolving play.doomland.xyz...
Found server play.doomland.xyz:25583 for domain play.doomland.xyz
Retrieving Server Info...
Server version : Doomland Network (protocol v763)
[MCC] Version is supported.
Logging in...
[MCC] Server is in offline mode.
[MCC] Server was successfully joined.
Type '/quit' to leave the server.
> Unhandled exception. System.IO.InvalidDataException: Failed to process incoming packet of type JoinGame. (PacketID: 40, Protocol: 763, LoginPhase: False, InnerException: System.Collections.Generic.KeyNotFoundException).
 ---> System.Collections.Generic.KeyNotFoundException: The given key 'minecraft:chat_type' was not present in the dictionary.
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at MinecraftClient.Protocol.Message.ChatParser.ReadChatType(Dictionary`2 registryCodec)
   at MinecraftClient.Protocol.Handlers.Protocol18Handler.HandlePacket(Int32 packetID, Queue`1 packetData)
   --- End of inner exception stack trace ---
   at MinecraftClient.Protocol.Handlers.Protocol18Handler.HandlePacket(Int32 packetID, Queue`1 packetData)
   at MinecraftClient.Protocol.Handlers.Protocol18Handler.Updater(Object o)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
Aborted (core dumped)

**Comment by milutinke** [comment]
Which server software are you running?

**Comment by david-cueto** [comment]
![image](https://github.com/MCCTeam/Minecraft-Console-Client/assets/80096375/6261a386-9b03-4682-bac8-30131f018fdf)

**Comment by milutinke** [comment]
> ![image](https://user-images.githubusercontent.com/80096375/283242203-6261a386-9b03-4682-bac8-30131f018fdf.png)

I meant like what server software, like Paper, or Spigot, or Vanilla?
And which version.

**Reply by david-cueto** [reply]
Oh sorry, 1.20.1 paper

**Reply by milutinke** [reply]
> Oh sorry, 1.20.1 paper

Can you send me the ip on Discord?
I will need to manually debug this.
Discord: @guyonahorse

**Reply by david-cueto** [reply]
It's strange because if I do it from my PC, it works, but from the VPS it gives the error, maybe it's because of the VPS.

**Comment by milutinke** [comment, marked as answer]
Figured it out
It's a custom proxy implementation, it does not send the chat type to the client
I tested it with 1.16.4 and 1.19.4 it works
